ast.go

```diff
@@ -1516,6 +1516,13 @@ func matchExactRegex(v string) ([]string, bool) {
 
 // matchRegex will match a regular expression to literals if possible.
 func matchRegex(re *syntax.Regexp) ([]string, bool) {
+
+	// Maximum number of literals that the expression should be expanded to. If
+	// this is exceeded, no expansion will be done. This allows reasonable
+	// optimizations of regex by expansion to literals but prevents cases
+	// where that expansion would result in a large number of literals.
+	const maxLiterals = 1000
```

Comment:
I think 100 is a better value. 1000 is likely too high.

Reply:
:+1: fixed